web-common/src/features/alerts/EditAlertDialog.svelte

```diff
@@ -90,6 +103,18 @@
       }
     },
   });
+  const { form } = formState;
+  $: if ($metricsViewSpec?.data && $timeRange?.data) {
+    const formValues = extractAlertFormValues(
+      queryArgsJson,
+      $metricsViewSpec.data,
+      $timeRange.data,
+    );
+    for (const fk in formValues) {
+      $form[fk] = formValues[fk];
+    }
+    console.log(formValues);
+  }
```

Comment:
Is this code necessary?

Reply:
Yes. Loading of metrics view spec and time range is async. So we need this to be loaded into the form. The other option is to get the args to the component, but this felt cleaner.